src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs

```diff
@@ -146,20 +146,45 @@ public override SyntaxNode VisitGroupClause(GroupClauseSyntax node)
 
                 by = by.Replace($"{result}.", string.Empty);
 
-                GroupByFields = by.Split(',');
+                var fields = by.Split(',');
 
-                for (int i = 0; i < GroupByFields.Length; i++)
+                var groupByFields = new List<string>(fields.Length);
+
+                for (var i = 0; i < fields.Length; i++)
                 {
-                    var field = GroupByFields[i];
+                    var parts = fields[i].Split('=');
 
-                    var parts = field.Split('=');
+                    string fieldName;
 
                     if (parts.Length == 1)
-                        GroupByFields[i] = parts[0].Trim();
+                        fieldName = parts[0].Trim();
                     else
-                        GroupByFields[i] = parts[1].Trim();
+                        fieldName = parts[1].Trim();
+
+                    if (fieldName.StartsWith('"') && fieldName.EndsWith('"'))
+                    {
+                        // group by "constant"
+                        continue;
+                    }
+
+                    if (char.IsNumber(fieldName[0]))
+                    {
+                        // group by 123
+                        continue;
+                    }
+
+                    if ("true".Equals(fieldName) || "false".Equals(fieldName))
+                    {
+                        // group by true
+                        // group by false
+                        continue;
+                    }
+
```

Comment:
Group by `null` ? 

Why do we have to check the field itself, can't we get access to the value toke directly?

Reply:
Right, another case.

We are checking it here because I recently added the following validation when we compile the static index:

https://github.com/ravendb/ravendb/blob/dc3a124e98472bebfc5dc29d43838be058a728e1/src/Raven.Server/Documents/Indexes/Static/IndexCompiler.cs#L370-L376